### ISSUE TITLE
English - Belgium not supported as culture identifier

### ISSUE THREAD

**Issue by Luc-Appelmans** [sql-bi/Bravo#324]
### Describe the problem

Error HTTP/500: Internal Server Error.
Culture is not supported. (Parameter 'culture')
4096 (0x1000) is an invalid culture identifier.

### Bravo version

0.9.3 (0.9.8139.22766)

### Tabular model

Power BI Desktop, import mode

### Power BI

Power BI Desktop version March 2022

### Steps to reproduce the behavior

1. Go to: Windows settings => Language & region 
2. Change the Regional format into **English (Belgium)**
3. Go to: Power BI Desktop => External Tools
4. Click on Bravo
5. You get the error!!

### Screenshots

![image](https://user-images.githubusercontent.com/103674484/163442986-025b76ee-e054-4227-b272-bea0e7e19aaf.png)

### Anything else ?

Trace Id: 00-434c2d5f08eb8961e9cb9087a6e8dcf3-9f8e07489e291bb8-00

**Comment by albertospelta** [comment]
I think we cannot solve this error because it is not raised by the application but by the Microsoft libraries that Bravo uses to communicate with the Power BI database server (Analysis Services). The culture "English (Belgium)" is identified by the LCID value 4096 which is [LOCAL_CUSTOM_UNSPECIFIED](https://docs.microsoft.com/en-us/windows/win32/intl/locale-custom-constants) and it is assigned to any culture that does not have complete system-provided data.

In this particular case I think the error is probably generated by trying to apply/translate into the `en-BE` culture another error message received from the server. Can you try to change the Windows display language to `en-US`? Does the application display a different error message?

Is this report a local Power BI Desktop report or is it connected to an online dataset? Can you please share a new and empty PBIX file created on your machine?

**Comment by Luc-Appelmans** [comment]
Hi Alberto,

Thanks for the fast reply.
I have changed the Windows display language to English (United States). But that did not remove the error.
The error only occurs when I use as Regional format English (Belgium), see print screen.
***@***.***

If I use as Regional format Dutch (Belgium) or French (Belgium), everything works fine (even in an empty file, see further).

I just installed Bravo yesterday and clicked on the Bravo button in External tools tab in Power BI Desktop.
The Power BI Desktop is completely empty, I just wanted to make sure it worked.

Doing some tests, I just noticed that if I start up Bravo in Power BI Desktop that has a model and some report pages, then I do not get the error 😊. Even with English (Belgium) as regional format.
But whenever the file is empty then I get the error.
Therefore, I would classify this as not at all urgent/important. Sorry for the disturbance.

But if you are interested to further investigate, I have attached the empty pbix files Bravo Test. This is completely empty and gives the error (probably because it is empty in combination with the English (Belgium) regional format)

Kind Regards,
Luc Appelmans
GSM: +32 473 110 150
***@***.******@***.***>

**Comment by albertospelta** [comment]
Hi @Luc-Appelmans, sorry but I'm not able to see any attachments, this is probably because replying by mail seems to get rid of any attachments. I think you have to reply through GitHub if you want to add a file to the thread. Thanks.

**Comment by Luc-Appelmans** [comment]
Hi Alberto,

Sorry for my late reply. Here is the file.
[Bravo Test.zip](https://github.com/sql-bi/Bravo/files/8535137/Bravo.Test.zip)
Though, as I mentioned, I only have the issue with an empty file.

Luc

**Comment by albertospelta** [comment]
Thanks @Luc-Appelmans! Finally managed to solve the error and the fix will be available in next release.

**Comment by albertospelta** [comment]
Fixed in version [0.9.4](https://github.com/sql-bi/Bravo/releases/tag/v0.9.4)

**Comment by albertospelta** [comment]
This issue only affects the .NET6 version of the framework and not the .NETFX/Core3.1/5.0 versions - see https://github.com/dotnet/runtime/issues/62929
There's a config setting we can use in .NET 6.0 to obtain the same behavior as in .NETFX/Core3.1/5.0:

```
<ItemGroup>
  <!-- Opt-out and use NLS instead of ICU globalization APIs -->
  <RuntimeHostConfigurationOption Include="System.Globalization.UseNls" Value="true" />
</ItemGroup>
```
